drivers/i2c/i2c_sam0.c

```diff
@@ -481,6 +481,9 @@ static int i2c_sam0_transfer(const struct device *dev, struct i2c_msg *msgs,
 		k_sem_take(&data->sem, K_FOREVER);
 
 		if (data->msg.status) {
+			/* return the bus to idle */
+			i2c->CTRLB.bit.CMD = 3;
```

Comment:
Why is the while() loop waiting on CLKHOLD not needed here but needed for the other cases below?

Reply:
This seems fair, and you've clearly done the hardware testing. +1